pkg/api/v1/clients/vault/resource_client.go

```diff
@@ -305,8 +307,12 @@ const (
 )
 
 func (rc *ResourceClient) resourceDirectory(namespace, directoryType string) string {
+	if rc.pathPrefix == "" {
```

Comment:
nit: I think it makes it easier to reason through if this default behavior is defined in the constructor. 

Also, the rootKey does not default in solo-kit and instead we accept whatever value is injected. Is there a particular reason we don't follow a similar pattern?

Reply:
Good callout. I was thinking of ensuring it at least defaults to `secret` since, in vault that's the behaviour when creating new clients. So if someone where to create a client connecting to a regular kv vault, they wouldn't have to specify it being "secret".